src/index.ts

```diff
@@ -0,0 +1,229 @@
+/*
+ * A stack-machine based Example syntax parser.
+ *
+ * Spec:
+ * 
+ *     A text is considered a named fragment when it is located between
+ *     "// BEGIN FRAGMENT: <name>" and "// END FRAGMENT", <> not included.
+ *     This syntax is case-sensitive.
+ * 
+ *     Don't use special characters in names. Prefer snake case.
+ * 
+ *     Fragments can be included in one another.
+ *     In that case the lines matching
+ *     "// BEGIN FRAGMENT: " and "// END FRAGMENT" are removed.
+ *     Elements between "// BEGIN ESCAPE" and "// END ESCAPE"
+ *     are excluded unconditionally from the tutorial.
+ * 
+ * Example code:
+ * 
+ *     import { ExampleParser } from "./example_parser"
+ *     let text: string = "// BEGIN FRAGMENT: test\nHello.\n// END FRAGMENT"
+ *     let parser_inst: ExampleParser = new ExampleParser(text)
+ *     console.log(parser_inst.mapLines())
+ * 
```

Comment:
I'd like to wrap it into `ts` code snippet:

```
* ```ts
* ...
* ```
```

Reply:
Thanks, done!